pilot/docker/wait-ready.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+url="${1:-localhost:15021/healthz/ready}"
+
+echo "Waiting for proxy to become ready..."
+while true; do
+  responseCode=$(curl -s --connect-timeout 1 -o /dev/null -w "%{http_code}" ${url})
+  if [ "$responseCode" == "200" ]; then
+    echo "Proxy is ready!"
+    exit 0
+  fi
+  echo "Proxy not ready (response code: ${responseCode})..."
```

Comment:
Unfortunately not. I thought about writing to a log file, but the container is typically configured with `readOnlyRootFilesystem: true`, so we'd need to add a volume, too.

Reply:
Bummer. The good news is if this is failing, the pod *should* also be failing readiness probe so the plain container log should have some info. There is a volume already mounted where we write the bootstrap config to that we could log I guess?

Reply:
Actually, I forgot to mention that when the hook fails, its output _is_ shown in the `PostStartHookError` event and the status shown in `kubectl get pods`. I'm adding a timeout to the hook so it fails after 60s (configurable; should default be 30s or even lower?). This way, if the hook fails, users can see what the problem is:
```
$ kubectl get po details-v1
NAME         READY   STATUS   RESTARTS   AGE
details-v1   1/2     PostStartHookError: command 'pilot-agent wait' exited with 255: Error: Timeout waiting for Envoy proxy to become ready. Last error: Get "http://localhost:15021/healthz/ready": dial tcp [::1]:15021: connect: connection refused
```

Reply:
What happens when it fails? Will it crashloop, and not just die? Because its not impossible that it will fail for 30s or so if pilot is temporarily down, etc so we will need some good recovery behavior. If it crashloops I think that's fine.. People who have sidecars but don't *need* them to function may be upset their apps are blocked but they can always opt out, and I think its a rare case

Reply:
If the container's post-start hook fails the container is terminated. The remaining containers are started. When the sidecar container is restarted, the post-start hook is invoked again. 

So, if the sidecar doesn't become ready, the application will nevertheless start after the timeout. After the sidecar container is restarted, the post-start hook runs again, but has no effect on the other containers (unless they have also terminated, in which case the hook delays their start once again).